viamonvif/service.go

```diff
@@ -175,21 +178,43 @@ func (dis *rtspDiscovery) DoCommand(ctx context.Context, command map[string]inte
 		if err != nil {
 			return nil, err
 		}
+
```

Comment:
Can we please refactor this into 1 method on `dis` with this signature:
```go
func (dis *rtspDiscovery) preview(ctx) (string, error) {
}
```
where `string` is `dataURL` 
And that method handles looking up the `snapshotURI`  and falling back to `fetchImageFromRTSPURL` if necessary?

That way this code becomes:
```
dataURL, err := dis.preview(ctx)
if err != nil {
return nil, err
}

return map[string]interface{}{"preview": dataURL}, nil
```

It also makes it much easier to move the preview functionality into a cli command if we want to.

Reply:
Good call, using `preview` helper now.

Everything was stuffed in the DoCommand because we only have one do-command handler.